docs/workflow/testing/host/testing.md

```diff
@@ -0,0 +1,79 @@
+# Building and running host tests
+
+The [host tests](/src/installer/tests) use [xunit](http://xunit.github.io/) for their testing framework.
+
+## Building tests
+
+To build the host tests, first build the product:
+
+1.  Build  CoreCLR and libraries:
+
+    * [CoreCLR](../../building/coreclr/README.md)
+    * [Libraries](../../building/libraries/README.md)
+    
+    Using the Release configuration of CoreCLR and libraries is recommended for running host tests.
+
+2.  Build the host and packs:
+    ```
+    build.cmd/sh -subset host+packs.product -runtimeConfiguration Release -librariesConfiguration Release
+    ```
+    If using a configuration other than Release for CoreCLR/libraries, specify the desired configuration in the `-runtimeConfiguration`/`-librariesConfiguration` arguments.
+
+### Building all tests
+
+The host tests are part of the `host` subset by default, so building the `host` subset also builds the host test. To build just the host tests:
+```
+build.cmd/sh -subset host.tests -runtimeConfiguration Release -librariesConfiguration Release
+```
+
+### Building specific tests
+
+A specific test project can also be directly built. For example:
+```
+dotnet build src\installer\tests\HostActivation.Tests
+```
+
+## Test context
+
+The host tests depend on:
+  1. Product binaries in a directory layout matching that of a .NET install
+  2. Restored [test projects](/src/installer/tests/Assets/TestProjects) which will be built and run by the tests
+  3. TestContextVariables.txt file with property and value pairs which will be read by the tests
+
+When [running all tests](#running-all-tests), the build is configured such that these are created/performed before the start of the test run.
+
+In order to create (or update) these dependencies without running all tests, the build targets that create them - RefreshProjectTestAssets and SetupTestContextVariables - can be directly run for the desired test project. For example:
+```
+dotnet build src\installer\tests\HostActivation.Tests -t:RefreshProjectTestAssets;SetupTestContextVariables -p:RuntimeConfiguration=Release -p:LibrariesConfiguration=Release
```

Comment:
It would probably be nice to add a subset that does this (similar to `libs.pretest`). But this at least documents what can be done the current system.